elements/lisk-p2p/test/unit/peer_pool.ts

```diff
@@ -71,14 +81,539 @@ describe('peerPool', () => {
 			seedPeers: [],
 			whitelisted: [],
 		},
+	};
+	let peerPool: PeerPool;
+	let peerInfo: P2PDiscoveredPeerInfo;
+	let nodeInfo: P2PNodeInfo;
+	let peerId: string;
+	let peerObject: any;
+	let messagePacket: any;
+	let requestPacket: any;
+	let clock: any;
+
+	beforeEach(async () => {
+		clock = sandbox.useFakeTimers();
+		peerPool = new PeerPool(peerPoolConfig);
+		peerId = '127.0.0.1:5000';
+		peerInfo = {
+			ipAddress: '127.0.0.1',
+			wsPort: 5000,
+			height: 1,
+			updatedAt: new Date(),
+			version: '1.0.1',
+			protocolVersion: '1.0.1',
+		};
+		nodeInfo = {
+			os: 'darwin',
+			version: '1.1',
+			protocolVersion: '1.0.1',
+			nethash: 'abc',
+			wsPort: 5000,
+			height: 1000,
+		};
+		requestPacket = { procedure: 'abc', data: 'abc' };
+		messagePacket = { ...requestPacket, event: 'abc' };
+		peerObject = {
+			...peerInfo,
+			id: peerId,
+			send: sandbox.stub(),
+			request: sandbox.stub(),
+			connect: sandbox.stub(),
+			applyPenalty: sandbox.stub(),
+			disconnect: sandbox.stub(),
+			removeListener: sandbox.stub(),
+			on: sandbox.stub(),
+			off: sandbox.stub(),
+			emit: sandbox.stub(),
+			destroy: sandbox.stub(),
+		} as any;
+		peerPool.emit = sandbox.stub().resolves();
+	});
+
+	afterEach(async () => {
+		clock.restore();
```

Comment:
We shouldn't have to do this.

Reply:
It is required otherwise the test hangs.

Reply:
You're right. But, perhaps the test failure when running this file standalone is related to this. Nevermind otherwise.